From cb41ef0d3f537f8d1fb345cf5bdcc30070333c8b Mon Sep 17 00:00:00 2001
From: Gert Wollny <gert.wollny@collabora.com>
Date: Mon, 27 Feb 2023 09:56:41 +0100
Subject: util/driconf: pin minImageCount to three for "Path of Exile"

"Path of Exile" will fail with an error "unsupported backbuffer image count"
when vkGetPhysicalDeviceSurfaceCapabilitiesKHR reports more than 3 as
minImageCount.

Signed-off-by: Gert Wollny <gert.wollny@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/21547>
---
 src/util/00-mesa-defaults.conf | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/util/00-mesa-defaults.conf b/src/util/00-mesa-defaults.conf
index 554115cedfc..843beae6609 100644
--- a/src/util/00-mesa-defaults.conf
+++ b/src/util/00-mesa-defaults.conf
@@ -891,6 +891,16 @@ TODO: document the other workarounds.
             <!-- https://gitlab.freedesktop.org/mesa/mesa/-/issues/5437 -->
             <option name="vk_dont_care_as_load" value="true" />
         </application>
+
+	<!-- Path of Exile stops with "unsupported backbuffer count" when more than
+	     3 swapchain image are reported. -->
+	<application name="Path of Exile" executable="PathOfExile_x64Steam.exe">
+	  <option name="vk_x11_override_min_image_count" value="3" />
+        </application>
+        <application name="Path of Exile" executable="PathOfExileSteam.exe">
+            <option name="vk_x11_override_min_image_count" value="3" />
+        </application>
+
     </device>
     <!-- vmwgfx doesn't like full buffer swaps and can't sync to vertical retraces.-->
     <device driver="vmwgfx">
-- 
cgit v1.2.1

